sherpa/astro/io/pyfits_backend.py

```diff
@@ -158,10 +158,10 @@ def _try_col(hdu, name, dtype=SherpaFloat, fix_type=False):
 
 
 def _try_tbl_col(hdu, name, dtype=SherpaFloat, fix_type=False):
-    if name not in hdu.columns.names:
-        return (None,)
-
-    col = hdu.data.field(name)
+    try:
+        col = hdu.data.field(name)
+    except KeyError:
+        return (None, )
```

Comment:
@hamogu this is essentially where the "magic" happens. Whereas before we did

    if name not in hdu.columns.names:
        hey column not found

we now rely on `hdu.data.field(name)` to deal with normalization of the column name. I haven't actually looked at the docs, but it performs case-insensitive access:

```
>>> from astropy.io import fits
>>> hdus = fits.open("1838_rprofile_rmid.fits")
>>> hdu = hdus[1]
>>> hdu.columns.names
WARNING: VerifyWarning: Invalid keyword for column 20: Coordinate/axis type option (TCTYPn) must be a string of atmost 8 characters (got 'BG_CEL_BRI'). The invalid keyword will be ignored for the purpose of formatting this column. [astropy.io.fits.column]
WARNING: VerifyWarning: Invalid keyword for column 21: Coordinate/axis type option (TCTYPn) must be a string of atmost 8 characters (got 'BG_CEL_BRI_ERR'). The invalid keyword will be ignored for the purpose of formatting this column. [astropy.io.fits.column]
WARNING: VerifyWarning: Invalid keyword for column 27: Coordinate/axis type option (TCTYPn) must be a string of atmost 8 characters (got 'CEL_BRI_ERR'). The invalid keyword will be ignored for the purpose of formatting this column. [astropy.io.fits.column]
['X', 'Y', 'RA', 'Dec', 'SHAPE', 'R', 'ROTANG', 'COMPONENT', 'AREA', 'EXPOSURE', 'COUNTS', 'ERR_COUNTS', 'COUNT_RATE', 'COUNT_RATE_ERR', 'BG_AREA', 'BG_EXPOSURE', 'BG_ERR', 'BG_COUNTS', 'BG_RATE', 'BG_SUR_BRI', 'BG_SUR_BRI_ERR', 'NET_COUNTS', 'NET_ERR', 'NET_RATE', 'ERR_RATE', 'SUR_BRI', 'SUR_BRI_ERR', 'RMID']
>>> hdu.data.field('Dec').sum()
-401.62284449878996
>>> hdu.data.field('DEC').sum()
-401.62284449878996
>>> hdu.data.field('dec').sum()
-401.62284449878996
>>> hdu.data.field('decx')
Traceback (most recent call last):
  File "/home/dburke/anaconda/envs/sherpa-main/lib/python3.9/site-packages/astropy/io/fits/column.py", line 2045, in _get_index
    indx = names.index(key.rstrip())
ValueError: 'decx' is not in list

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/dburke/anaconda/envs/sherpa-main/lib/python3.9/site-packages/astropy/io/fits/fitsrec.py", line 691, in field
    column = self.columns[key]
  File "/home/dburke/anaconda/envs/sherpa-main/lib/python3.9/site-packages/astropy/io/fits/column.py", line 1643, in __getitem__
    key = _get_index(self.names, key)
  File "/home/dburke/anaconda/envs/sherpa-main/lib/python3.9/site-packages/astropy/io/fits/column.py", line 2054, in _get_index
    raise KeyError(f"Key '{key}' does not exist.")
KeyError: "Key 'decx' does not exist."
>>>
```

Reply:
Fantastic! I know I was missing something.